src/touchhandler/TouchHandler.cpp

```diff
@@ -50,46 +50,77 @@ Pinetime::Applications::TouchEvents TouchHandler::GestureGet() {
+  // The last event is a e.g. "slide" event with info.touching == true.
+  // gestureReleased state does not have to be computed manually, instead it occurs when event != "none".
+
+  // GESTURE configurations (P8a variants) of the fused (usually) Cst716 generate no events during the physical gesture.
+  // The only event is a e.g. "slide" event with info.touching == true.
+  // gestureReleased state does not have to be computed manually, instead it occurs everytime.
+
+  // DYNAMIC configurations (PineTime) are configured in reporting mode during initialisation.
+  // Usually based on the Cst816s, they generate multiple e.g. "slide" gesture events with info.touching == true during the physical gesture.
+  // The last of these e.g. "slide" events has info.touching == false.
+  // gestureReleased state is computed manually by checking for the transition to info.touching == false.
 
-  if (info.gesture != Pinetime::Drivers::Cst816S::Gestures::None) {
-    if (gestureReleased) {
-      if (info.gesture == Pinetime::Drivers::Cst816S::Gestures::SlideDown ||
-          info.gesture == Pinetime::Drivers::Cst816S::Gestures::SlideLeft ||
-          info.gesture == Pinetime::Drivers::Cst816S::Gestures::SlideUp ||
-          info.gesture == Pinetime::Drivers::Cst816S::Gestures::SlideRight ||
-          info.gesture == Pinetime::Drivers::Cst816S::Gestures::LongPress) {
-        if (info.touching) {
+  // Unfortunately, there is no way to reliably obtain which configuration is used at runtime.
+  // In all cases, the event is bubbled up once the gesture is released.
+
+  #if defined(DRIVER_TOUCH_REPORT)
+    if (info.gesture != Pinetime::Drivers::Cst816S::Gestures::None) {
+      gesture = ConvertGesture(info.gesture);
+      info.touching = false;
+    }
+  #elif defined(DRIVER_TOUCH_GESTURE) 
+    if (info.gesture != Pinetime::Drivers::Cst816S::Gestures::None) {
+      gesture = ConvertGesture(info.gesture);
+    }
+  #elif defined(DRIVER_TOUCH_DYNAMIC)
+    if (info.gesture != Pinetime::Drivers::Cst816S::Gestures::None) {
+      if (gestureReleased) {
+        if (info.gesture == Pinetime::Drivers::Cst816S::Gestures::SlideDown ||
+            info.gesture == Pinetime::Drivers::Cst816S::Gestures::SlideLeft ||
+            info.gesture == Pinetime::Drivers::Cst816S::Gestures::SlideUp ||
+            info.gesture == Pinetime::Drivers::Cst816S::Gestures::SlideRight ||
+            info.gesture == Pinetime::Drivers::Cst816S::Gestures::LongPress) {
+          if (info.touching) {
+            gesture = ConvertGesture(info.gesture);
+            gestureReleased = false;
+          }
+        } else {
           gesture = ConvertGesture(info.gesture);
-          gestureReleased = false;
         }
-      } else {
-        gesture = ConvertGesture(info.gesture);
       }
     }
-  }
 
-  if (!info.touching) {
-    gestureReleased = true;
-  }
+    if (!info.touching) {
+      gestureReleased = true;
+    }
+  #endif
```

Comment:
We try to avoid using preprocessor. These different drivers hardly have any shared code, so they should probably be separate classes with a common interface.

Reply:
I agree that it would be nice to avoid using preprocessor statements, however I don't know how useful it would actually be to split this piece of code into some kind of abstraction. It is essentially the same driver, only the data it reports is slightly different for each chip.

In any case, we can continue the discussion in https://github.com/InfiniTimeOrg/InfiniTime/pull/1130 .